doc/man/man3/notcurses_tabbed.3.md

```diff
@@ -0,0 +1,159 @@
+% notcurses_tabbed(3)
+% v2.2.3
+
+# NAME
+
+notcurses_tabbed – tabbed interface widget
+
+# SYNOPSIS
+
+**#include <notcurses/notcurses.h>**
+
+```c
+#define NCTABBED_OPTION_BOTTOM 0x0001
+
+struct nctabbed;
+struct ncplane;
+struct nctab;
+
+typedef struct nctabbed_options {
+  uint64_t selchan; // channel for the selected tab header
+  uint64_t hdrchan; // channel for unselected tab headers
+  uint64_t sepchan; // channel for the tab separator
+  char* separator;  // separator string (copied by nctabbed_create())
+  uint64_t flags;   // bitmask of NCTABBED_OPTION_*
+} nctabbed_options;
+
+typedef void (*tabcb)(struct nctab* t, struct ncplane* ncp, void* userptr);
+```
+
+**struct nctabbed* nctabbed_create(struct ncplane* ***ncp***, const nctabbed_options* ***opts***);**
+
+**void nctabbed_destroy(struct nctabbed* ***nt***);**
+
+**void nctabbed_redraw(struct nctabbed* ***nt***);**
+
+**void nctabbed_ensure_selected_header_visible(struct nctabbed* ***nt***);**
+
+**struct nctab* nctabbed_selected(struct nctabbed* ***nt***);**
+
+**struct nctab* nctabbed_leftmost(struct nctabbed* ***nt***);**
+
+**int nctabbed_tabcount(struct nctabbed* ***nt***);**
+
+**struct ncplane* nctabbed_plane(struct nctabbed* ***nt***);**
+
+**struct ncplane* nctabbed_content_plane(struct nctabbed* ***nt***);**
+
+**const char* nctab_name(struct nctab* ***t***);**
+
+**int nctab_name_width(struct nctab* ***t***);**
+
+**void* nctab_userptr(struct nctab* ***t***);**
+
+**struct nctab* nctab_next(struct nctab* ***t***);**
+
+**struct nctab* nctab_prev(struct nctab* ***t***);**
+
+**struct nctab* nctabbed_add(struct nctabbed* ***nt***, struct nctab* ***after***, struct nctab* ***before***, tabcb ***tcb***, const char* ***name***, void* ***opaque***);
+
+**int nctabbed_del(struct nctabbed* ***nt***, struct nctab* ***t***);**
+
+**int nctab_move(struct nctabbed* ***nt***, struct nctab* ***t***, struct nctab* ***after***,
+                   struct nctab* ***before***);**
+
+**void nctab_move_right(struct nctabbed* ***nt***, struct nctab* ***t***);**
+
+**void nctab_move_left(struct nctabbed* ***nt***, struct nctab* ***t***);**
+
+**void nctabbed_rotate(struct nctabbed* ***nt***, int ***amt***);**
+
+**struct nctab* nctabbed_next(struct nctabbed* ***nt***);**
+
+**struct nctab* nctabbed_prev(struct nctabbed* ***nt***);**
+
+**struct nctab* nctabbed_select(struct nctabbed* ***nt***, struct nctab* ***t***);**
+
+**void nctabbed_channels(struct nctabbed* ***nt***, uint64_t* RESTRICT ***hdrchan***, uint64_t* RESTRICT ***selchan***, uint64_t* RESTRICT ***sepchan***);**
+
+**uint64_t nctabbed_hdrchan(struct nctabbed* ***nt***);**
+
+**uint64_t nctabbed_selchan(struct nctabbed* ***nt***);**
+
+**uint64_t nctabbed_sepchan(struct nctabbed* ***nt***);**
+
+**const char* nctabbed_separator(struct nctabbed* ***nt***);**
+
+**int nctabbed_separator_width(struct nctabbed* ***nt***);**
```

Comment:
ahhh you even got the variable three-star bullshit, yay, thanks, i know this is the most boring task in the world

Reply:
life without multicursor would be a nightmare...

Reply:
i'm a vim guy. visual studio code did this for you? nice. makes me proud to work there.

Reply:
I tried using VSCode at first but code linting for complex projects (in C) is trash so I ended up using CLion which works well with CMake and also has multicursor support.